### ISSUE TITLE
SnapshotCollector is showing when app is running in Release configuration

### ISSUE THREAD

**Issue by bbougot** [Microsoft/ApplicationInsights-dotnet#638]
I am using Microsoft.ApplicationInsights.SnapshotCollector 1.0.10 for my WPF .NET 4.6.2 application. The workflow is working fine, but the issue I encounter is the MinidumpUploader.exe console is showing when my app (in Release mode) is running. 

I would like to know how to hide the MinidumpUploader.exe console, at least when my app is running in Release configuration.

**Comment by bbougot** [comment]
I have decompiled Microsoft.ApplicationInsights.SnapshotCollector and in the Uploader.cs file I've juste seen: 

![capture](https://user-images.githubusercontent.com/8962802/31861545-670d79ba-b72f-11e7-8867-f03facb64880.JPG)

If MinidumpUploader could not be found in the shadow copy folders, why don't the process start being hidden?

This should be used:
        Process.Start(new ProcessStartInfo()
        {
          FileName = processPath,
          Arguments = arguments,
          CreateNoWindow = true,
          UseShellExecute = false
        });

**Comment by weng5e** [comment]
Hi,

Thank you so much for reporting the issue. Please send Snapshot related issues to snapshothelp@microsoft.com. I will reply to your email.

Thanks!
Wuyi

**Comment by pharring** [comment]
@bbougot thanks for the report and the email.

Line 120 in the decompiled code is a "last resort" attempt to start the MinidumpUploader. The fact that you got there indicates a problem in the preceding code which tries to find a suitable shadow-copy location, copy the binaries there and launching the uploader. A side-effect of this last resort code-path is to show the console window. It's an accidental canary for a bug in the shadow-copy logic.

We're working on a few fixes:
1. Add more telemetry around this, so we can detect how common it is.
2. As you suggest, avoid showing the console on the 'last resort' path.
3. Diagnose and fix the underlying bug with shadow-copy. We've already identified one issue where we're comparing UTC with local time stamps. There may be more to it.

Hopefully, this will be fixed in the next few days and pushed to Nuget. (Sadly, only the first of them made it into the 1.0.11 release today).

**Comment by bbougot** [comment]
@pharring
Thanks for your answer.

Indeed, there is comparison of LastWrite access DateTime between files being located in the SnapshotCollector folder and those which are copied in the shadow folder (/Temp for me). 

I had to use a dirty fix to solve this. I set the LastWrite time access to DateTime.Max to my local SnapshotCollector files at launch of my app, so that the ShadowCopy process think the local files should be copied into the ShadowCopy filter and used from it. Otherwise, the process don’t use the ShadowCopy folder and execute the MinidumpUploader located inside the root folder of the app (where it does not even exist because MinidumpUploader is located in /SnapshotCollector folder)
